app/views/hostgroups/_form.html.erb

```diff
@@ -1,5 +1,5 @@
 <%= javascript 'host_edit', 'host_edit_interfaces', 'class_edit' %>
-<%= form_for @hostgroup, :html => {:data => {:id => @hostgroup.try(:id), :submit => 'progress_bar' }} do |f| %>
+<%= form_for @hostgroup, :html => {:data => {:id => @hostgroup.try(:id), :submit => 'progress_bar', :"history-url" => session["redirect_to_url_#{controller_name}"] }} do |f| %>
```

Comment:
Shouldn't we you the same method as we have in `cancel_path` later in the form? I guess it should not be called `cancel_path_with_index_search`, as it's not about cancel anymore, but using the same method can ensure we are consistent in the future.

Reply:
Yes , agreed on using same method. I will replace fetching from session part with this method by renaming it.